crates/rspack_binding_options/src/options/raw_module/js_value_ref.rs

```diff
@@ -0,0 +1,74 @@
+use std::{
+  marker::PhantomData,
+  sync::{Arc, Mutex},
+};
+
+use napi::bindgen_prelude::*;
+use napi::{Env, NapiValue, Ref};
+
+pub struct JsValueRef<T: NapiValue> {
+  env: Env,
+  ref_: Arc<Mutex<Ref<()>>>,
+  _phantom: PhantomData<T>,
+}
+
+unsafe impl<T: NapiValue> Send for JsValueRef<T> {}
```

Comment:
This data structure is not `Send` and `Sync`. As you can not access `env` and the data behind this on the other thread.